.pre-commit-config.yaml

```diff
@@ -33,35 +33,35 @@ repos:
     hooks:
       - id: mypy
         args: [--ignore-missing-imports]
-        additional_dependencies: [
-          types-toml,
-          types-pkg_resources,
-          types-chardet,
-          types-appdirs,
-          types-colorama,
-          types-pyyaml,
-          types-regex,
-        ]
+        additional_dependencies:
+          [
+            types-toml,
+            types-pkg_resources,
+            types-chardet,
+            types-appdirs,
+            types-colorama,
+            types-pyyaml,
+            types-regex,
+          ]
         files: ^src/sqlfluff/.*
   - repo: https://github.com/pycqa/flake8
     rev: 4.0.1
     hooks:
       - id: flake8
-        additional_dependencies: [
-          flake8-black>=0.2.4,
-          flake8-docstrings,
-        ]
+        additional_dependencies: [flake8-black>=0.2.4, flake8-docstrings]
   - repo: https://github.com/pycqa/doc8
     rev: 0.10.1
     hooks:
       - id: doc8
-        args: [
-          --file-encoding,
-          utf8,
-        ]
+        args: [--file-encoding, utf8]
         files: docs/source/.*\.rst$
   - repo: https://github.com/adrienverge/yamllint.git
     rev: v1.26.3
     hooks:
       - id: yamllint
         args: [-c=.yamllint]
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
```

Comment:
The README for ruff advertises:
> [Near-parity](https://github.com/charliermarsh/ruff#how-does-ruff-compare-to-flake8) with the built-in Flake8 rule set

In light of that, should we remove flake8 (line 50 above)? It would be odd to run two highly overlapping style checker tools.

If we consider doing this, we should make sure something is still enforcing/running Black at commit time. You'll see references to flake8 and Black in this file, also in `requirements_dev.txt`, possibly other places.

Reply:
For now I was thinking just having them to run together, since Ruff runs nearly instantly, and it doesn't introduce thrash (they don't try to change things back and forth). What do you think?

Reply:
There's no harm to adding it, so I'm happy with it. Just pointing out it may not add a lot of value because of the overlap. I like trying new tools, though! In fact, I had already run ruff on one of my projects -- can't remember which one.